op-chain-ops/genesis/check.go

```diff
@@ -66,46 +139,101 @@ func CheckPredeploys(db vm.StateDB) error {
 			return fmt.Errorf("no code found at predeploy %s", addr)
 		}
 
+		if UntouchablePredeploys[addr] {
+			continue
+		}
+
 		// There must be an admin
 		admin := db.GetState(addr, AdminSlot)
 		adminAddr := common.BytesToAddress(admin.Bytes())
 		if addr != predeploys.ProxyAdminAddr && addr != predeploys.GovernanceTokenAddr && adminAddr != predeploys.ProxyAdminAddr {
-			return fmt.Errorf("admin is %s when it should be %s for %s", adminAddr, predeploys.ProxyAdminAddr, addr)
+			return fmt.Errorf("expected admin for %s to be %s but got %s", addr, predeploys.ProxyAdminAddr, adminAddr)
 		}
 	}
 
 	// For each predeploy, check that we've set the implementation correctly when
 	// necessary and that there's code at the implementation.
 	for _, proxyAddr := range predeploys.Predeploys {
-		implAddr, special, err := mapImplementationAddress(proxyAddr)
-		if err != nil {
-			return err
+		if UntouchablePredeploys[*proxyAddr] {
+			continue
+		}
+
+		if *proxyAddr == predeploys.LegacyERC20ETHAddr {
+			continue
 		}
 
-		if !special {
-			impl := db.GetState(*proxyAddr, ImplementationSlot)
-			implAddr := common.BytesToAddress(impl.Bytes())
-			if implAddr == (common.Address{}) {
-				return fmt.Errorf("no implementation for %s", *proxyAddr)
+		var expImplAddr common.Address
+		var err error
+		if *proxyAddr == predeploys.ProxyAdminAddr {
+			expImplAddr = predeploys.ProxyAdminAddr
+		} else {
+			expImplAddr, err = AddressToCodeNamespace(*proxyAddr)
+			if err != nil {
+				return fmt.Errorf("error converting to code namespace: %w", err)
 			}
 		}
 
-		implCode := db.GetCode(implAddr)
+		implCode := db.GetCode(expImplAddr)
 		if len(implCode) == 0 {
 			return fmt.Errorf("no code found at predeploy impl %s", *proxyAddr)
 		}
+
+		if expImplAddr == predeploys.ProxyAdminAddr {
+			continue
```

Comment:
Can this continue be hoisted to where the assignment happens instead of having this branch below?